src/LumicalSD.cc

```diff
@@ -57,51 +57,43 @@ void LumicalSD::Initialize(G4HCofThisEvent* hce)
 // --- core of the SD, running whenever a step is inside the assigned volume(s) ------
 G4bool LumicalSD::ProcessHits(G4Step* step, G4TouchableHistory*)
 {
-    //const G4StepPoint* endPoint = step->GetPostStepPoint(); 
+    //const G4StepPoint* endPoint = step->GetPostStepPoint();
     // check that an real interaction occured (eg. not a transportation)
     //G4StepStatus stepStatus = endPoint->GetStepStatus();
     //G4bool transmit = (stepStatus==fGeomBoundary || stepStatus==fWorldBoundary);
     //if (transmit) return true;
 
     //-----------create a hit and populate it with the information ----------
-        
-    G4double edep = (step->GetTotalEnergyDeposit()/MeV) - (step->GetNonIonizingEnergyDeposit()/MeV);
+
+    // G4double edep = (step->GetTotalEnergyDeposit()/MeV) - (step->GetNonIonizingEnergyDeposit()/MeV);
+    G4double edep = (step->GetTotalEnergyDeposit()/MeV);
     if (edep <= 0.) return true;
-    
-        
+
+
     G4StepPoint* preStepPoint = step->GetPreStepPoint();
     G4TouchableHistory* touchable = (G4TouchableHistory*)(preStepPoint->GetTouchable());
-    G4ThreeVector position = step->GetTrack()->GetPosition();    // position in mm
-   
+    // G4ThreeVector position = step->GetTrack()->GetPosition();    // position in mm
+
     //G4int plan = touchable->GetCopyNumber(3);
     G4int plan = touchable->GetReplicaNumber(2);
     G4int zone = touchable->GetReplicaNumber(1);
     G4int pad  = touchable->GetReplicaNumber(0);
-   
+
     G4int padID = pad + 64 * zone + 256 * ((plan-33)/15);
     LumicalHit *hit = (*fHitsCollection)[padID];
 
-    hit->totene += edep;
-    
-    //G4ThreeVector position = preStepPoint->GetPosition();
-    hit->x = position.getX();
-    hit->y = position.getY();
-    hit->z = position.getZ();
-
-    G4ThreeVector momentum = preStepPoint->GetMomentum();
-    hit->px = momentum.getX();
-    hit->py = momentum.getY();
-    hit->pz = momentum.getZ();
-    
-     
-    hit->SetEdep(edep); 
-    //fHitsCollection -> insert(hit);
-      
-    return true;
-   
-}
-
+    hit->fEdep += edep;
 
+    G4ThreeVector position = preStepPoint->GetPosition();
```

Comment:
I assume this is commented accidentally?
Previously you was taking track position as a hit position instead of preStepPoint position. I am not sure if they are the same..